wandb/apis/public.py

```diff
@@ -2394,16 +2394,32 @@ def update_variables(self):
         self.variables.update({"cursor": self.cursor})
 
     def convert_objects(self):
-        return [
-            Artifact(
-                self.client,
-                self.run.entity,
-                self.run.project,
-                r["node"]["digest"],
-                r["node"],
-            )
-            for r in self.last_response["project"]["run"][self.run_key]["edges"]
-        ]
+        results = []
+        for r in self.last_response["project"]["run"][self.run_key]["edges"]:
```

Comment:
So in the artifact query, we're actually pulling down the version index and the collection name:

```
fragment ArtifactFragment on Artifact {
    id
    ...
    versionIndex
    artifactSequence {
        id
        name
    }
    ...
}
```

Using `artifactSequence.name` and `versionIndex` we can always generate its `name:alias`. This might be slightly cleaner than assembling it from the list of aliases.

Reply:
Just slightly ;)